### PR TITLE
[2.x] Adds support for guzzlehttp/promises 2.0

### PR DESCRIPTION
**Pull request by joedixon**
We previously had to lock `guzzlehttp/promises` at `^1.4.0` because an upstream dependency started to allow `^1.4.0||^2.0`.

We were using the `Promise\settle()` function which was deprecated and removed in `2.0`.

I refactored the code to use `Utils::settle()` which is supported in both 1.4 and 2.0 so I believe we're safe setting this to `^1.4.0||^2.0` to ensure it doesn't break for anyone relying on `1.4`.

I also removed `psr/http-message` as I can't find it used anywhere in the package.

### CODE REVIEW

**Comment by scottwakefield** [comment]
Hey, @joedixon 👋 

We're seeing a lot of the following errors in our application logs on Vapor, making it hard to parse our logs for errors in our code 😅:

```
{
    "exception":  {
        "class": "Error",
        "message": "Call to undefined function GuzzleHttp\Promise\settle()",
        "code": 0,
        "file": "/var/task/vendor/laravel/vapor-core/src/Runtime/Handlers/WarmerHandler.php:25"
    }
}
```

Is that likely related to this? If so, do you know when this PR might get merged?

**Comment by joedixon** [comment]
Hey @scottwakefield, if you update to the latest version of vapor-core, you should already see the error is resolved as we've fixed guzzle/promises to 1.4.

This PR, when merged will also provide support for version 2.

**Comment by scottwakefield** [comment]
Ah, perfect! Thanks for the quick reply, @joedixon 👍🏻

**Comment by gdebrauwer** [comment]
Any idea when this might get merged? This is preventing me from updating to vapor-core v2.33.1, which also means I can not fix the RDS certificate expiry issue on my vapor databases.

**Comment by joedixon** [comment]
Hey @gdebrauwer that's just been merged and tagged
